app/src/main/java/org/mozilla/fenix/home/collections/TabInCollection.kt

```diff
@@ -0,0 +1,300 @@
+}
+
+/**
+ * Composable used to display the background of a [Tab] shown in collections that is being swiped left or right.
+ *
+ * @param dismissDirection [DismissDirection] of the tab being swiped depending on which this composable
+ * will also indicate the swipe direction by placing a warning icon at the start of the swipe gesture.
+ * If `null` the warning icon will be shown at both ends.
+ * @param isLastInCollection Whether the tab is to be shown between others or as the last one in collection.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun DismissedTabBackground(
+    dismissDirection: DismissDirection?,
+    isLastInCollection: Boolean,
+) {
+    Card(
+        modifier = Modifier.fillMaxSize(),
+        backgroundColor = FirefoxTheme.colors.layer3,
+        shape = if (isLastInCollection) bottomTabShape else middleTabShape,
+        elevation = 0.dp,
+    ) {
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Icon(
+                painter = painterResource(drawable.ic_delete),
+                contentDescription = null,
+                modifier = Modifier
+                    .padding(horizontal = 32.dp)
+                    // Only show the delete icon for where the swipe starts.
+                    .alpha(
+                        if (dismissDirection == StartToEnd) 1f else 0f
+                    ),
+                tint = FirefoxTheme.colors.iconWarning,
+            )
+
+            Icon(
+                painter = painterResource(drawable.ic_delete),
+                contentDescription = null,
+                modifier = Modifier
+                    .padding(horizontal = 32.dp)
+                    // Only show the delete icon for where the swipe starts.
+                    .alpha(
+                        if (dismissDirection == EndToStart) 1f else 0f
+                    ),
+                tint = FirefoxTheme.colors.iconWarning,
+            )
+        }
+    }
+}
+
+/**
+ * Clips the Composable this applies to such that it cannot draw content / shadows outside it's top bound.
+ */
+private fun Modifier.clipTop() = this.then(
+    Modifier.drawWithContent {
+        val paddingPx = Constraints.Infinity.toFloat()
+        clipRect(
+            left = 0f - paddingPx,
+            top = 0f,
+            right = size.width + paddingPx,
+            bottom = size.height + paddingPx
+        ) {
+            this@drawWithContent.drawContent()
+        }
+    }
+)
+
+/**
+ * Get a friendlier short url for [url].
+ *
+ * @param url Full url to be shortened.
+ *
+ * @see toShortUrl
+ */
+@Composable
+private fun shortenUrl(url: String): String {
+    return if (inComposePreview) {
+        url
+    } else {
+        url.toShortUrl(LocalContext.current.components.publicSuffixList)
+    }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
```

Comment:
```suggestion
@Composable
@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO)
```

instead of doing two duplicate preview blocks for light/dark, it's possible to do two consecutive preview calls.

Reply:
Didn't know that. Awesome tip, thank you!

Reply:
Tried it now exactly as written here and I see two previews with both using light mode though even after "Build Refresh".
Could be needing a newer Android Studio?
Will remove one version nonetheless.

Reply:
Is the theme still being set to light in the `FirefoxTheme` initialization?

Reply:
😅 That was it.
Removing it I see errors in the preview rendering:
> java.lang.ClassCastException: class com.android.layoutlib.bridge.android.BridgeContext cannot be cast to class org.mozilla.fenix.FenixApplication (com.android.layoutlib.bridge.android.BridgeContext is in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @195c4744; org.mozilla.fenix.FenixApplication is in unnamed module of loader org.jetbrains.android.uipreview.ModuleClassLoader @283fdca)   at org.mozilla.fenix.ext.ContextKt.getApplication(Context.kt:29)   at org.mozilla.fenix.ext.ContextKt.getComponents(Context.kt:35)   at org.mozilla.fenix.ext.ContextKt.settings(Context.kt:57)   at org.mozilla.fenix.theme.Theme$Companion.getTheme(FirefoxTheme.kt:40)

Checked the codebase, seems like this cool double preview of the themed composable isn't used anywhere else so maybe we can't use it just yet :-(.

Reply:
It's because of a limitation in `FirefoxTheme` currently, so the initialization needs to be:

`FirefoxTheme(theme = Theme.getTheme(isPrivate = false)) {`

Reply:
Worked! Thank you!

Reply:
I think I remember that limitation being because of the compose version used? And I think now that can be avoided?